Automattic/wp-calypso
# BackupBrowser: improve error handling

The purpose of this PR is to improve error handling in the backup browser.

## Proposed Changes
* Add a `notices.ts` file with all the notices and messages.
* Add two additional retries when attempting to retrieve a file info (`path-info`) and throw an error notice in case of failure.
* Add additional validations when trying to get the download URL, so we could stop the loading indicator and throw an error notice.
* Improve error handling when there's an error fetching the URL to preview a file. In this scenario, we are not adding an error notice. It will just stop loading.

## Error notices
| Scenario | Message screenshot |
|---|---|
| User tries to prepare and download a database table | <img width="538" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1488641/16eebd1e-e396-43ac-bb8f-f094f3edb427"> |
| User tries to download a file | <img width="551" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1488641/86a4f9c7-b96f-4e71-a9e9-442c1d93e1bf"> |
| User clicks on a file, and it fails to fetch file info | <img width="575" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1488641/4a73efc9-eae0-4457-8bb1-3dd724693766"> |

## Testing Instructions
* Navigate through the backup browser and ensure everything is still working as usual.
* The best way is to trigger errors is sandboxing the WPCOM API and force an error when:
  * Attempting to fetch a file information (`path-info`)
  * Attempting to prepare and download a database table (`filtered/prepare`)
  * Attempting to preview a file (`file stream url`)

## Pre-merge Checklist
